src/query/api/v1/handler/namespace/schema_test.go

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package namespace
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/m3db/m3/src/cluster/kv"
+	nsproto "github.com/m3db/m3/src/dbnode/generated/proto/namespace"
+	"github.com/m3db/m3/src/dbnode/namespace"
+	"github.com/m3db/m3/src/dbnode/namespace/kvadmin"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	mainProtoStr = `syntax = "proto3";
+
+package mainpkg;
+
+import "mainpkg/imported.proto";
+
+message TestMessage {
+  double latitude = 1;
+  double longitude = 2;
+  int64 epoch = 3;
+  bytes deliveryID = 4;
+  map<string, string> attributes = 5;
+  ImportedMessage an_imported_message = 6;
+}
+`
+	importedProtoStr = `
+syntax = "proto3";
+
+package mainpkg;
+
+message ImportedMessage {
+  double latitude = 1;
+  double longitude = 2;
+  int64 epoch = 3;
+  bytes deliveryID = 4;
+}
+`
+	testSchemaJQ = `
+{
+  name: "testNamespace",
+  msgName: "mainpkg.TestMessage",
+  protoName: "mainpkg/test.proto",
+  protoMap: 
+    {
+      "mainpkg/test.proto": $file1,
+      "mainpkg/imported.proto": $file2
+    }
+}
+`
+	testSchemaJson = `
```

Comment:
Is there any way to generate this programatically so its not such a pain to edit? Does json.Marshal() not handle escaping?

Reply:
If you look at the test closely, you will find that I have code to generate it, genTestJson, this testSchemaJson is generated by genTestJson.

I used jq to generate it, but jq is not installed on buildkite, so even though the test succeeds locally, it always fail on buildkite. I documented it in the test.

Reply:
oh I see. So if you want to update this variable you should just run that test? Can you add a comment above this line saying: "Run blah blah blah to regenerate this string"